test/docker/setup.sh

```
@@ -424,7 +424,26 @@ else
-run_quiet docker restart boltz
+# Provision the Ark wallet (fulmine) BEFORE restarting boltz. Boltz's Rust sidecar (boltzr)
+# attaches the ARK wallet only if it can connect to fulmine once, at startup; that connection has
+# no retry. If fulmine has no wallet yet when boltzr boots, the ARK currency is left wallet-less
+# and /v2/swap/restore fails forever with "no wallet for ARK". So create+unlock fulmine first,
+# then restart boltz so boltzr re-runs that one-shot connection against a now-ready fulmine.
 provision_boltz_fulmine
+run_quiet docker restart boltz
+
+echo "waiting for Boltz to be ready after restart..."
+boltz_ready=0
+for i in {1..30}; do
```

Shellcheck (0.11.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)